python/tests/test_aggregates.py

```diff
@@ -0,0 +1,335 @@
+                                            AVG(DISTINCT c3) AS c3,
+                                            AVG(DISTINCT c4) AS c4,
+                                            AVG(DISTINCT c5) AS c5,
+                                            AVG(DISTINCT c6) AS c6,
+                                            AVG(DISTINCT c7) AS c7,
+                                            AVG(DISTINCT c8) AS c8
+                                       FROM avg0_distinct_groupby
+                                       GROUP BY id;''')
+        data = [
+            {"id" :0 ,"c1": 1, "c2": 2, "c3": 3, "c4": 4, "c5": 5, "c6": 6, "c7": None, "c8": 8},
+            {"id" :0 ,"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"id": 1,"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"id": 1,"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        
+        out = sql.listen("avg_view_distinct_groupby")
+        
+        sql.start()
+
+        sql.input_json("avg0_distinct_groupby", data)
+        
+        expected_data = [
+            {"id": 0, "c1": 1, "c2": 2, "c3": 16, "c4": 4, "c5": 5, "c6": 33, "c7": 70, "c8": 8},
+            {"id" : 1, "c1": 12, "c2": 21, "c3" : 44, "c4" : 38, "c5" : 42, "c6" : 48, "c7":50, "c8": 77}]
+        
+        sql.wait_for_completion(True)
+        
+        out_data = out.to_dict()
+        
+        for datum in expected_data:
+            datum.update({"insert_delete": 1})
+        
+        assert expected_data == out_data
+        
+    def test_avg0_where(self):
+        sql = SQLContext('test_avg0_where', TEST_CLIENT).get_or_create()
+        
+        sql.register_table_from_sql(''' 
+                                    CREATE TABLE avg0_where(
+                                        c1 TINYINT,
+                                        c2 TINYINT NOT NULL,
+                                        c3 INT2,
+                                        c4 INT2 NOT NULL,
+                                        c5 INT,
+                                        c6 INT NOT NULL,
+                                        c7 BIGINT,
+                                        c8 BIGINT NOT NULL);''')
+        
+        sql.register_materialized_view("avg_view_where", '''
+                                       SELECT
+                                            AVG(c1) AS c1,
+                                            AVG(c2) AS c2,
+                                            AVG(c3) AS c3,
+                                            AVG(c4) AS c4,
+                                            AVG(c5) AS c5,
+                                            AVG(c6) AS c6,
+                                            AVG(c7) AS c7,
+                                            AVG(c8) AS c8
+                                       FROM avg0_where
+                                       WHERE 
```

Comment:
here I would also add a different predicate - so two instances of this test.

Reply:
Does this mean I should create a different test or add one more condition?

Reply:
Let's do the other refactoring first